pkg/mic/mic.go

```diff
@@ -462,6 +441,32 @@ func (c *Client) RemoveAssignedIDsWithDeps(assignedID *aadpodid.AzureAssignedIde
 	return nil
 }
 
-func (c *Client) GetAssignedIDName(podName string, podNameSpace string, idName string) string {
+func (c *Client) getAssignedIDName(podName string, podNameSpace string, idName string) string {
 	return podName + "-" + podNameSpace + "-" + idName
 }
+
+func (c *Client) convertIDListToMap(arr *[]aadpodid.AzureIdentity) (m map[string]aadpodid.AzureIdentity, err error) {
+	m = make(map[string]aadpodid.AzureIdentity)
```

Comment:
specify map size to `len(arr)` to avoid multiple allocs below.

Reply:
good catch.. thank you